tex/homology/cup-product.tex

```diff
@@ -89,6 +91,60 @@ \section{de Rham cohomology}
 So the failure of $\alpha$ to be exact is signaling
 that $H_1(S^1) \cong \ZZ$.
 
+As another piece of intuition, note that:
+\begin{itemize}
+	\ii each $k$-differential form $\omega$ can be interpreted as a function that takes each
+	$k$-smooth submanifold $S \subseteq M$, and returns a real number $\int_S \omega$.
+	\ii let us pretend that all $k$-simplices are smooth for now. Then we have:
+	\begin{itemize}
+		\ii The $k$-cochains are the functions that sends each $k$-simplex to a real number.
+		\ii The $k$-cocycles are the $k$-cochains that sends the boundaries to $0$.
+		\ii The $k$-coboundaries are the $k$-cochains that sends the cycles to $0$.
+	\end{itemize}
+
+	Meanwhile:
+	\begin{itemize}
+		\ii The differential forms are the functions that sends each $k$-simplex to a real number,
+		satisfying certain linearity and smoothness properties --- for instance:
+		\begin{itemize}
+			\ii if $k \geq 1$ and a $k$-simplex has the image contained in a point, then it must be
+			sent to $0$;
+			\ii if we reparametrize a $k$-simplex, the assigned value must be the same;
+			\ii if we flip two vertices of a $k$-simplex, the assigned value must be negated;
+			\ii if a $k$-simplex can be formed by gluing two $k$-simplices along a face,
+			then the assigned value must be the sum of the corresponding values assigned to the
+			sub-$k$-simplices;
+			\ii etc.
```

Comment:
This part is a bit ugly, that it nests 3 layers of itemize... but I don't know how to better write it. Maybe just delete the innermost block if it doesn't make sense/doesn't help much in understanding?

Reply:
I think if you need the nesting you should just use it without worrying about exactly how the typesetting turns out, it seems to make sense here.